docs/api-docs/headless/end-to-end-guide/graphql.mdx

```diff
@@ -0,0 +1,752 @@
+# End-to-End Guide: Headless Checkout Flow with GraphQL APIs
+
+BigCommerce provides GraphQL Storefront APIs used by server-side and browser-side clients for managing cart and checkout objects to build storefront features. The GraphQL APIs offer the same use cases as our Cart and Checkout REST Management APIs but have the added benefit of making it easier to build headless storefront applications. Developers using GraphQL can spend less time focusing on API interactions and more time building the shopper experience.
+
+This end-to-end guide demonstrates how to power headless storefronts and other non-Stencil shopper experiences. These steps will allow you to build shopping experiences where the purchase funnel is entirely powered by GraphQL, from catalog browsing through checkout. The examples walk you through critical GraphQL capabilities to fit most use cases when creating a headless storefront:
+
+* [creating a token](#create-a-customer-impersonation-token)
+* [querying product data](#querying-product-data)
+* [creating a cart](#creating-a-cart)
+* [adding cart line items](#add-cart-line-items)
+* [getting a Checkout](#get-a-checkout)
+* [adding a checkout billing address](#add-a-checkout-billing-address)
+* [adding a checkout shipping consignment](#add-a-checkout-shipping-consignment)
+* [selecting shipping option](#select-checkout-shipping-option)
+* [completing a checkout](#complete-checkout) 
+
+
+<Callout type="info">
+You can use the GraphQL Storefront Playground to perform the example queries presented in this guide. To access the GraphQL Storefront API Playground and documentation, [sign in to your store](https://login.bigcommerce.com/deep-links/manage), and navigate to **Settings** > **API** > **Storefront API Playground**.
+</Callout>
+
+## Prerequisites
+
+To complete the guide, you will need the following:
+* [BigCommerce sandbox store](https://start.bigcommerce.com/developer-sandbox/) 
+* Familiarity with creating a channel and a site
+
+## Accessing Storefront GraphQL APIs
+You can make BigCommerce GraphQL requests in the context of the server to server API. The server-side API clients have the option of using either a Storefront API token or a customer impersonation token to access Storefront GraphQL APIs. This tutorial describes how to create and use a customer impersonation token to authenticate and access the Storefront GraphQL APIs. 
+
+### Create a customer impersonation token
+
+A customer impersonation token permits you to perform actions as any shopper. To generate a [customer impersonation token](/docs/rest-authentication/tokens#customer-impersonation-tokens), send a POST request to `/v3/storefront/api-token-customer-impersonation`.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+ ```http filename="Example request: Create a customer impersonation token"
+POST https://api.bigcommerce.com/stores/{STORE_HASH}/v3/storefront/api-token-customer-impersonation
+X-Auth-Token: {{ACCESS_TOKEN}}
```

Comment:
🍹 Small nitpick - does the number of curly braces matter? I notice we have `{STORE_HASH}` and `{{ACCESS_TOKEN}}`